static/js/browser2/observation_chart.tsx

```diff
@@ -0,0 +1,273 @@
+        .call(d3.axisBottom(x).tickFormat(d3.timeFormat("%b %e")));
+    } else {
+      svg
+        .append("g")
+        .attr("transform", `translate(0,${HEIGHT})`)
+        .call(d3.axisBottom(x));
+    }
+    if (plotData[0].value < 10) {
+      // TODO(chejennifer): figure out a way to remove sub ticks.
+      svg.append("g").call(d3.axisLeft(y).ticks(4).tickFormat(d3.format("d")));
+    } else {
+      // The number of decimals depends on the domain range compared with the
+      // Maximum value. When the range is narrow, need more decimals to display
+      // the difference.
+      const domainDiff = y.domain()[1] - y.domain()[0];
+      const decimal = Math.ceil(Math.log10(y.domain()[1] / (domainDiff / 4)));
+      svg.append("g").call(
+        d3
+          .axisLeft(y)
+          .ticks(4)
+          .tickFormat(d3.format(`.${decimal}s`))
+      );
+    }
+
+    // draw y label.
+    const yLabelText = this.getYLabelText();
+    svg
+      .append("text")
+      .attr("transform", "rotate(-90)")
+      .attr("y", 0 - LEFT)
+      .attr("x", 0 - HEIGHT / 2)
+      .attr("dy", "1em")
+      .style("text-anchor", "middle")
+      .text(yLabelText);
+
+    this.addTooltip();
+  }
+
+  private addTooltip(): void {
+    d3.select("#svg-container" + this.props.idx)
+      .attr("style", "position: relative")
+      .append("div")
+      .attr("id", TOOLTIP_ID)
+      .attr("style", "position: absolute; display: none; z-index: 1");
+  }
+
+  private getYLabelText(): string {
+    let yLabelText = "";
+    if (this.props.sourceSeries["unit"]) {
+      yLabelText = this.props.sourceSeries["unit"];
+    }
+    if (
+      this.props.sourceSeries["scalingFactor"] &&
+      this.props.sourceSeries["scalingFactor"] === "100"
+    ) {
+      yLabelText = "%";
+    }
+    return yLabelText;
+  }
+
+  private handleDotHover = (dataPoint: any): void => {
+    const tooltipSelect = d3
+      .select("#svg-container" + this.props.idx)
+      .select(`#${TOOLTIP_ID}`);
+    const text =
+      dataPoint.dateString + ": " + dataPoint.value + this.getYLabelText();
+    const tooltipHeight = (tooltipSelect.node() as HTMLDivElement).clientHeight;
+    const offset = 10;
+    const leftOffset = offset;
+    const topOffset = -tooltipHeight - offset;
+    tooltipSelect
+      .text(text)
+      .style("left", d3.event.offsetX + leftOffset + "px")
+      .style("top", d3.event.offsetY + topOffset + "px")
+      .style("display", "block");
+  };
+
+  private handleDotLeave = (): void => {
+    d3.select("#svg-container" + this.props.idx)
+      .select(`#${TOOLTIP_ID}`)
+      .style("display", "none");
+  };
+
+  private handleDotClick = (dataPoint: any): void => {
+    // TODO (chejennifer): might need better way of getting dcid of observation node because this can be very slow
+    const date = dataPoint.dateString;
+    let request = `/api/browser/observationId?place=${this.props.placeDcid}&stat_var=${this.props.statVarId}&date=${date}`;
```

Comment:
can use "observation-id" and "stat-var" in url:

Reply:
hm looks like we've been a little inconsistent with how we're specifying these. would be nice to add this to the developer guide (or start a new style guide)

so i took a look and what we have is mostly:
- routes - dash separated (except for legacy paths)
- get params - camel case

Reply:
done

& I can add this to the developer guide